### PR TITLE
Don't retry failed compactions.

### PR DESCRIPTION
**Pull request by Gouthamve**
Fixes prometheus/prometheus#3487

@fabxc

---
This change is [Reviewable](https://reviewable.io/reviews/prometheus/tsdb/207)

### CODE REVIEW

**Comment by Gouthamve** [comment]
Running this on my Prometheus server now. Nothing broken. Will keep an eye, but good to go for a review.

File: compact.go [hunk 1]
```diff
@@ -360,16 +384,20 @@ func (w *instrumentedChunkWriter) WriteChunks(chunks ...ChunkMeta) error {
 func (c *LeveledCompactor) write(dest string, meta *BlockMeta, blocks ...BlockReader) (err error) {
 	level.Info(c.logger).Log("msg", "compact blocks", "count", len(blocks), "mint", meta.MinTime, "maxt", meta.MaxTime)
 
+	var tmp string
+
```
Comment (fabxc) [inline compact.go]: Maybe just move the defer below setting of `dir` and `tmp`.

Comment (Gouthamve) [reply compact.go]: Done.

**Comment by imkin** [comment]
What is the release timeframe for this fix?

**Comment by Gouthamve** [comment]
@imkin I am hoping to do a release by the end of the week.

**Pull request merged**